services/download/src/test/java/app/coronawarn/server/services/download/DownloadIntegrationTest.java

```diff
@@ -172,16 +174,14 @@ void testDownloadRunSuccessfully() {
     assertThat(federationBatchInfoRepository.findByStatus("ERROR")).hasSize(1);
     assertThat(federationBatchInfoRepository.findByStatus("ERROR_WONT_RETRY")).hasSize(1);
 
-    Iterable<DiagnosisKey> diagnosisKeys = diagnosisKeyRepository.findAll();
+    final Set<String> diagnosisKeys = StreamSupport
+        .stream(diagnosisKeyRepository.findAll().spliterator(), false)
+        .collect(Collectors.groupingBy(d -> ByteString.copyFrom(d.getKeyData()).toStringUtf8()))
+        .keySet();
     assertThat(diagnosisKeys)
         .hasSize(3)
-        .contains(createDiagnosisKey(BATCH1_DATA))
-        .contains(createDiagnosisKey(BATCH2_DATA))
-        .contains(createDiagnosisKey(RETRY_BATCH_SUCCESSFUL_DATA));
-  }
-
-  private DiagnosisKey createDiagnosisKey(String keyData) {
-    return DiagnosisKey.builder()
-        .fromFederationDiagnosisKey(FederationBatchTestHelper.createDiagnosisKey(keyData)).build();
```

Comment:
Why do we need this change?

Reply:
I thought it would be good to not create the diagnosis keys objects again but rather use their keys instead for testing.

Reply:
@sorin-stefan-iovita-sap / @mibrasap - be very careful, with what this change all implies....
What does `contains` do on a `Set<String>` vs. `Iterable<DiagnosisKey>`? I assume at some point it will for sure use `equals` - once on String and the other implementation on DiagnosisKey!!! As well as it might depend on the implementation of the Set vs. Iterable... the access methods might differ... see: most Sets are using HashSet as implementation - contains, now first checks based upon the has value, if a bucket exists with the same hash and only in case this is true, walks through the whole bucket and checks on each item in it the equals method. In comparison to a list, where the Iterable will start at one end of the List and checks only equals on all objects, till one is found. 

Long explanation short: what do you want to achieve by this refactoring? Generating new objects within unit tests is quite common and doesn't harm, since the JVM is anyhow short living - this can't be the reason.
Tests have to be maintained, code is a lot more often read than written... try to keep it simple. Make sure it's easy to understand what's tested and which productive code is covered by the test. (equals? hashCode?)

Reply:
Allright! Then let's keep the things simple. I'll leave it as it was before 😄  .